source/Octopus.Tentacle.Tests.Integration/PollingTentacleTest.cs

```diff
@@ -64,32 +60,28 @@ public async Task BasicCommunicationsWithWithAnOldPollingTentacle()
             octopus.Trust(Support.Certificates.TentaclePublicThumbprint);
 
             var cts = new CancellationTokenSource();
-            var tentacleId = PollingSubscriptionId.Generate();
 
             using var tmp = new TemporaryDirectory();
             var oldTentacleExe = await TentacleFetcher.GetTentacleVersion(tmp.DirectoryPath, "6.3.451");
 
-            var (disposable, runningTentacle) = new PollingTentacleBuilder(port, Support.Certificates.ServerPublicThumbprint)
-                .WithTentaclePollSubscription(tentacleId)
-                .WithTentacleExe(oldTentacleExe)
-                .Build(cts.Token);
-
-            using (disposable)
+            using (var runningTentacle = new PollingTentacleBuilder(port, Support.Certificates.ServerPublicThumbprint)
+                       .WithTentacleExe(oldTentacleExe)
+                       .Build(cts.Token))
             {
                 var testTask = Task.Run(() =>
                 {
                     var tentacleClient = new TentacleClientBuilder(octopus)
                         .WithRemoteThumbprint(Support.Certificates.TentaclePublicThumbprint)
-                        .WithServiceUri(tentacleId)
+                        .WithServiceUri(runningTentacle.ServiceUri)
                         .Build(CancellationToken.None);
 
                     var res = tentacleClient.ScriptService.GetStatus(new ScriptStatusRequest(new ScriptTicket("1212"), 111));
                     res.ExitCode.Should().Be(0);
                 }, cts.Token);
 
-                await Task.WhenAny(runningTentacle, testTask);
+                await Task.WhenAny(runningTentacle.Task, testTask);
                 cts.Cancel();
-                await Task.WhenAll(runningTentacle, testTask);
+                await Task.WhenAll(runningTentacle.Task, testTask);
```

Comment:
I think I might do what is suggested [here](https://github.com/OctopusDeploy/OctopusTentacle/pull/470/files#r1185710292) to remove the need to do this little dance.